File: TOMIMSV4/DAL/Sistema_Jornada/clsLnVW_Fecha_Recepcion_TMS_Ticket.vb
Imports System.Data.SqlClient

Public Class clsLnVW_Fecha_Recepcion_TMS_Ticket

    Public Shared Sub Cargar(ByRef oBeTMP_VW_Fecha_Recepcion_TMS_Ticket As clsBeVW_Fecha_Recepcion_TMS_Ticket, ByRef dr As DataRow)

        Try

            With oBeTMP_VW_Fecha_Recepcion_TMS_Ticket
                .IdTicket = IIf(IsDBNull(dr.Item("IdTicket")), 0, dr.Item("IdTicket"))
                .IdOrdenCompraEnc = IIf(IsDBNull(dr.Item("IdOrdenCompraEnc")), 0, dr.Item("IdOrdenCompraEnc"))
                .IdRecepcionEnc = IIf(IsDBNull(dr.Item("IdRecepcionEnc")), 0, dr.Item("IdRecepcionEnc"))
                .Fecha_Ingreso = IIf(IsDBNull(dr.Item("Fecha_Ingreso")), Date.Now, dr.Item("Fecha_Ingreso"))
                .Fecha_Creacion = IIf(IsDBNull(dr.Item("Fecha_Creacion")), Date.Now, dr.Item("Fecha_Creacion"))
                .Fecha_Recepcion = IIf(IsDBNull(dr.Item("Fecha_Recepcion")), Date.Now, dr.Item("Fecha_Recepcion"))
            End With

        Catch ex As Exception
            Throw ex
        End Try

    End Sub

    Public Shared Function Get_All() As List(Of clsBeVW_Fecha_Recepcion_TMS_Ticket)

        Dim lReturnList As New List(Of clsBeVW_Fecha_Recepcion_TMS_Ticket)

        Try

            Const sp As String = "SELECT * FROM TMP_VW_Fecha_Recepcion_TMS_Ticket "

            Using lConnection As New SqlConnection(connectionString:=Configuration.ConfigurationManager.AppSettings("CST"))

                lConnection.Open()

                Using lTransaction As SqlTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUncommitted)

                    Using lDTA As New SqlDataAdapter(sp, lConnection)

                        lDTA.SelectCommand.CommandType = CommandType.Text
                        lDTA.SelectCommand.Transaction = lTransaction
                        Dim lDataTable As New DataTable
                        lDTA.Fill(lDataTable)

                        Dim vBeTMP_VW_Fecha_Recepcion_TMS_Ticket As New clsBeVW_Fecha_Recepcion_TMS_Ticket

                        For Each dr As DataRow In lDataTable.Rows
                            vBeTMP_VW_Fecha_Recepcion_TMS_Ticket = New clsBeVW_Fecha_Recepcion_TMS_Ticket()
                            Cargar(vBeTMP_VW_Fecha_Recepcion_TMS_Ticket, dr)
                            lReturnList.Add(vBeTMP_VW_Fecha_Recepcion_TMS_Ticket)
                        Next

                    End Using

                    lTransaction.Commit()

                End Using

                lConnection.Close()

            End Using

            Return lReturnList

        Catch ex As Exception
            Throw ex
        End Try

    End Function

    Public Shared Sub GetSingle(ByRef pBeTMP_VW_Fecha_Recepcion_TMS_Ticket As clsBeVW_Fecha_Recepcion_TMS_Ticket)

        Try

            Const sp As String = "SELECT * FROM TMP_VW_Fecha_Recepcion_TMS_Ticket"


            Using lConnection As New SqlConnection(connectionString:=Configuration.ConfigurationManager.AppSettings("CST"))

                lConnection.Open()

                Using lTransaction As SqlTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUncommitted)

                    Using lDTA As New SqlDataAdapter(sp, lConnection)

                        lDTA.SelectCommand.CommandType = CommandType.Text
                        lDTA.SelectCommand.Transaction = lTransaction
                        Dim lDataTable As New DataTable
                        lDTA.Fill(lDataTable)

                        Dim vBeTMP_VW_Fecha_Recepcion_TMS_Ticket As New clsBeVW_Fecha_Recepcion_TMS_Ticket

                        If lDataTable IsNot Nothing AndAlso lDataTable.Rows.Count > 0 Then
                            Cargar(vBeTMP_VW_Fecha_Recepcion_TMS_Ticket, lDataTable.Rows(0))
                        End If

                    End Using

                    lTransaction.Commit()

                End Using

                lConnection.Close()

            End Using

        Catch ex As Exception
            Throw ex
        End Try

    End Sub

End Class

File: WMS.DALCore/Ticket/clsLnVW_Fecha_Recepcion_TMS_Ticket.cs
namespace WMS.DALCore.Ticket
{
    using System;
    using System.Data;
    using System.Collections.Generic;
    using WMS.EntityCore.Ticket;
    using Microsoft.Data.SqlClient;
    using Microsoft.Extensions.Configuration;

    public class clsLnVW_Fecha_Recepcion_TMS_Ticket
    {
        public static void Cargar(ref clsBeVW_Fecha_Recepcion_TMS_Ticket oBeTMP_VW_Fecha_Recepcion_TMS_Ticket, DataRow dr)
        {
            try
            {
                oBeTMP_VW_Fecha_Recepcion_TMS_Ticket.IdTicket = (dr["IdTicket"] == DBNull.Value) ? 0 : Convert.ToInt32(dr["IdTicket"]);
                oBeTMP_VW_Fecha_Recepcion_TMS_Ticket.IdOrdenCompraEnc = (dr["IdOrdenCompraEnc"] == DBNull.Value) ? 0 : Convert.ToInt32(dr["IdOrdenCompraEnc"]);
                oBeTMP_VW_Fecha_Recepcion_TMS_Ticket.IdRecepcionEnc = (dr["IdRecepcionEnc"] == DBNull.Value) ? 0 : Convert.ToInt32(dr["IdRecepcionEnc"]);
                oBeTMP_VW_Fecha_Recepcion_TMS_Ticket.Fecha_Ingreso = (dr["Fecha_Ingreso"] == DBNull.Value) ? DateTime.Now : Convert.ToDateTime(dr["Fecha_Ingreso"]);
                oBeTMP_VW_Fecha_Recepcion_TMS_Ticket.Fecha_Creacion = (dr["Fecha_Creacion"] == DBNull.Value) ? DateTime.Now : Convert.ToDateTime(dr["Fecha_Creacion"]);
                oBeTMP_VW_Fecha_Recepcion_TMS_Ticket.Fecha_Recepcion = (dr["Fecha_Recepcion"] == DBNull.Value) ? DateTime.Now : Convert.ToDateTime(dr["Fecha_Recepcion"]);
            }
            catch (Exception)
            {
                throw;
            }
        }

        public static List<clsBeVW_Fecha_Recepcion_TMS_Ticket> Get_All(IConfiguration config)
        {
            List<clsBeVW_Fecha_Recepcion_TMS_Ticket> lReturnList = new List<clsBeVW_Fecha_Recepcion_TMS_Ticket>();

            try
            {
                const string sp = "SELECT * FROM TMP_VW_Fecha_Recepcion_TMS_Ticket ";

                using (SqlConnection lConnection = new SqlConnection(config.GetConnectionString("CST")))
                {
                    lConnection.Open();

                    using (SqlTransaction lTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUncommitted))
                    {
                        using (SqlDataAdapter lDTA = new SqlDataAdapter(sp, lConnection))
                        {
                            lDTA.SelectCommand.CommandType = CommandType.Text;
                            lDTA.SelectCommand.Transaction = lTransaction;
                            DataTable lDataTable = new DataTable();
                            lDTA.Fill(lDataTable);

                            clsBeVW_Fecha_Recepcion_TMS_Ticket vBeTMP_VW_Fecha_Recepcion_TMS_Ticket;

                            foreach (DataRow dr in lDataTable.Rows)
                            {
                                vBeTMP_VW_Fecha_Recepcion_TMS_Ticket = new clsBeVW_Fecha_Recepcion_TMS_Ticket();
                                Cargar(ref vBeTMP_VW_Fecha_Recepcion_TMS_Ticket, dr);
                                lReturnList.Add(vBeTMP_VW_Fecha_Recepcion_TMS_Ticket);
                            }
                        }

                        lTransaction.Commit();
                    }

                    lConnection.Close();
                }

                return lReturnList;
            }
            catch (Exception)
            {
                throw;
            }
        }

        public static void GetSingle(IConfiguration config, ref clsBeVW_Fecha_Recepcion_TMS_Ticket pBeTMP_VW_Fecha_Recepcion_TMS_Ticket)
        {
            try
            {
                const string sp = "SELECT * FROM TMP_VW_Fecha_Recepcion_TMS_Ticket";

                using (SqlConnection lConnection = new SqlConnection(config.GetConnectionString("CST")))
                {
                    lConnection.Open();

                    using (SqlTransaction lTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUncommitted))
                    {
                        using (SqlDataAdapter lDTA = new SqlDataAdapter(sp, lConnection))
                        {
                            lDTA.SelectCommand.CommandType = CommandType.Text;
                            lDTA.SelectCommand.Transaction = lTransaction;
                            DataTable lDataTable = new DataTable();
                            lDTA.Fill(lDataTable);

                            clsBeVW_Fecha_Recepcion_TMS_Ticket vBeTMP_VW_Fecha_Recepcion_TMS_Ticket = new clsBeVW_Fecha_Recepcion_TMS_Ticket();

                            if (lDataTable != null && lDataTable.Rows.Count > 0)
                            {
                                var lrow = lDataTable.Rows[0];
                                Cargar(ref vBeTMP_VW_Fecha_Recepcion_TMS_Ticket, lrow);
                            }
                        }

                        lTransaction.Commit();
                    }

                    lConnection.Close();
                }
            }
            catch (Exception)
            {
                throw;
            }
        }
    }
}